public/activity.html

```
@@ -190,405 +173,285 @@ <h3 class="font-bold text-lg mb-4 flex items-center">
+            </div>
+          </aside>
+
+          <div class="lg:col-span-2 space-y-6">
+            <!-- Not Joined CTA -->
+            <div id="join-cta" class="hidden rounded-2xl p-8 bg-gradient-to-br from-teal-50 to-cyan-50 dark:from-teal-900/20 dark:to-cyan-900/20 border border-teal-200 dark:border-teal-800 text-center">
+              <div class="text-5xl mb-4">🔒</div>
+              <h3 class="font-bold text-2xl text-gray-800 dark:text-gray-200 mb-2">Join to Access Details</h3>
+              <p class="text-gray-600 dark:text-gray-300 text-sm mb-6 max-w-sm mx-auto">
+                Session locations and descriptions are encrypted and only revealed to participants.
+              </p>
+              <button id="btn-join" onclick="joinActivity()"
```

HTMLHint (1.9.2)

[warning] The type attribute must be present on <button> elements.

(button-type-require)